kotsadm/operator/pkg/client/client.go

```diff
@@ -231,38 +231,41 @@ func (c *Client) registerHandlers(socketClient *socket.Client) error {
 
 		log.Println("received a deploy request for", args.AppSlug)
 
-		var applyResult, helmResult *commandResult
-		var applyError, helmError error
+		var deployRes *deployResult
+		var dryRunResult, applyResult, helmResult *commandResult
+		var deployError, helmError error
 		defer func() {
-			if applyError != nil {
-				applyResult = &commandResult{
-					hasErr:      true,
-					multiStdout: [][]byte{},
-					multiStderr: [][]byte{[]byte(applyError.Error())},
-				}
+			// only send results if we have an error
+			if deployRes.dryRunResult.hasErr {
+				dryRunResult = &deployRes.dryRunResult
 			}
-			if helmError != nil {
-				helmResult = &commandResult{
-					hasErr:      true,
-					multiStdout: [][]byte{},
-					multiStderr: [][]byte{[]byte(helmError.Error())},
-				}
+			if deployRes.applyResult.hasErr {
+				applyResult = &deployRes.applyResult
+			}
+			if helmError == nil {
+				helmResult = &commandResult{}
 			}
 
-			err := c.sendResult(args, nil, applyResult, helmResult)
+			err := c.sendResult(args, dryRunResult, applyResult, helmResult)
 			if err != nil {
 				log.Printf("failed to report result: %v", err)
 			}
 		}()
 
-		applyResult, applyError = c.deployManifests(args)
-		if applyError != nil {
-			log.Printf("falied to deploy manifests: %v", applyError)
+		deployRes, deployError = c.deployManifests(args)
+		if deployError != nil {
+			deployRes = &deployResult{}
+			deployRes.applyResult.hasErr = true
+			deployRes.applyResult.multiStderr = [][]byte{[]byte(deployError.Error())}
+			log.Printf("falied to deploy manifests: %v", deployError)
 			return
```

Comment:
```suggestion
			return nil
```

Reply:
Does this makes sense if we don't have return argument for this function?

Reply:
oh, no it doesn't 🤦 . github diff did not help there for sure.